cfme/storage/volume_backup.py

```diff
@@ -0,0 +1,266 @@
+
+        return (
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(VolumeBackupDetailsToolbar)
+    sidebar = View.nested(VolumeBackupDetailSidebar)
+    entities = View.nested(VolumeBackupDetailsEntities)
+
+
+class VolumeRestoreView(VolumeBackupView):
+    """The restore Volume backup view"""
+    @property
+    def is_displayed(self):
+        return False
+
+    volume_name = BootstrapSelect(name='volume_id')
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('cancel')
+
+
+@attr.s
+class VolumeBackup(BaseEntity, WidgetasticTaggable):
+    """ Model of an Storage Volume Backups in cfme
+
+    Args:
+        name: name of the backup
+        provider: provider
+    """
+    name = attr.ib()
+    provider = attr.ib()
+
+    def restore(self, name):
+        """Restore the volume backup. this feature included in 5.9 and above.
+
+        Args:
+            name: volume name
+        """
+        view = navigate_to(self, 'Restore')
+        view.volume_name.fill(name)
+        view.save.click()
+        view.flash.assert_success_message('Restoring Cloud Volume "{}"'.format(self.name))
+
+    def refresh(self):
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except BackupNotFound:
+            return False
+
+    @property
+    def size(self):
+        """ size of cloud volume backup.
+
+        Returns:
+            :py:class:`int' size of volume backup.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Size')
+
+    @property
+    def status(self):
+        """ Present status of cloud volume backup.
+
+        Returns:
+            :py:class:`str' status [available/restoring] of volume backup.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of('Status')
+
+    @property
+    def volume(self):
+        """ volume name of backup.
+
+        Returns:
+            :py:class:`str' respective volume name.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud Volume')
+
+
+@attr.s
+class VolumeBackupCollection(BaseCollection):
+    """Collection object for :py:class:'cfme.storage.volume_backups.VolumeBackup' """
+    ENTITY = VolumeBackup
+
+    def all(self):
+        """returning all backup objects for respective storage manager type"""
+        view = navigate_to(self, 'All')
+
+        backups = [self.instantiate(name=item['Name'], provider=self.filters.get('provider'))
+                   for item in view.entities.elements.read()
+                   if self.filters.get('provider').name in item['Cloud Provider']]
```

Comment:
REQUIRED: according to pep 8 list comprehensions shouldn't take several lines or be complex.  Please convert this to regular for statement